crates/config/src/config.rs

```diff
@@ -184,6 +186,103 @@ impl Default for OffChainValidatorConfig {
     }
 }
 
+/// Config the base/minimal fee rules
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct FeeConfig {
+    meta_contract_fee_weight: u8,
+    sudt_transfer_fee_weight: u8,
+    withdraw_fee_weight: u8,
+    /// HashMap<fee_sudt_id, fee_rate>
+    ///
+    /// fee_rate is known as gasPrice in Ethereum.
+    ///
+    /// The defaut fee_sudt_id is 1, which is CKB, and the ckb_fee_rate is denoted in shannons,
+    /// which itself is a fractional denomination of CKBytes.
+    ///   1 CKByte = 100,000,000 Shannons
+    ///
+    /// Note: The key of all TOML maps must be strings.
+    fee_rates: HashMap<Uint32, u64>,
+}
+impl FeeConfig {
+    pub fn is_supported_sudt(&self, sudt_id: u32) -> bool {
+        if self.fee_rates.is_empty() {
+            return true;
+        }
+        self.fee_rates.contains_key(&sudt_id.into())
+    }
+    pub fn get_fee_rate(&self, sudt_id: u32) -> Result<u128, Error> {
+        if self.fee_rates.is_empty() {
+            return Ok(0);
+        }
+        let fee_rate = self
+            .fee_rates
+            .get(&sudt_id.into())
+            .ok_or(Error::UnsupportedFeeSudt)?;
+        Ok(fee_rate.to_owned().into())
+    }
+    /// Get the minimal fee of meta contract
+    pub fn meta_contract_base_fee(&self, sudt_id: u32) -> Result<u128, Error> {
+        let fee_rate = self.get_fee_rate(sudt_id)?;
+        Ok(fee_rate * u128::from(self.meta_contract_fee_weight))
+    }
+    /// Get the minimal fee of a native sudt transfer transaction
+    pub fn sudt_transfer_base_fee(&self, sudt_id: u32) -> Result<u128, Error> {
+        let fee_rate = self.get_fee_rate(sudt_id)?;
+        Ok(fee_rate * u128::from(self.sudt_transfer_fee_weight))
+    }
+    /// Get the minimal fee of a withdrawal request
+    pub fn withdrawal_base_fee(&self, sudt_id: u32) -> Result<u128, Error> {
+        let fee_rate = self.get_fee_rate(sudt_id)?;
+        Ok(fee_rate * u128::from(self.withdraw_fee_weight))
+    }
+    /// Get the minimal gasPrice of Polyjuice contract
+    pub fn polyjuice_base_gas_price(&self, sudt_id: u32) -> Result<u128, Error> {
+        self.get_fee_rate(sudt_id)
+    }
+}
+impl Default for FeeConfig {
+    /// The suggested configs:
+    /// ```toml
+    /// [mem_pool.fee_config]
+    /// meta_contract_fee_weight = 2
+    /// sudt_transfer_fee_weight = 2
+    /// withdraw_fee_weight = 2
+    ///
+    /// [mem_pool.fee_config.fee_rates]
+    /// 0x1 = 500
+    /// ```
+    fn default() -> Self {
+        Self {
+            meta_contract_fee_weight: 2,
```

Comment:
Why the default weight is 2, do we have any reason?

Reply:
No, just a simple small default value.
Do you have any suggestion?

Reply:
We can just use `#[derive(Default)]` on this structure.